src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -212,19 +212,31 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
             throw new NotNormalizableValueException(sprintf('Could not denormalize object of type "%s", no supporting normalizer found.', $type));
         }
 
+        $notNormalizableExceptions = [];
         if (isset($context[DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS])) {
+            if ($context[DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS]) {
```

Comment:
```suggestion
        if ($context[DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS] ?? false) {
```
To avoid the two levels of "if"?

And then we can put
```php
unset($context[DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS]);
```
outside the if, as in either cases, `$context[DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS]` won't be set at the end

The same goes for the `COLLECT_EXTRA_ATTRIBUTES_ERRORS` part